### PR TITLE
enhancement: download vc redist on windows if latest is not installed

### LINKED ISSUE #4250: [Task]: Create a script to install latest VS C++ Redistributables
### Task description

Windows users with outdated VS C++ tools reported issues with loading the native rust modules into Firefly. This resulted in an app crash on startup.

Bug report: https://github.com/iotaledger/firefly/issues/4113

To mitigate this, there should be an nsis script included in FF, to prompt the windows users to install the latest VS C++ Redistributables.

Additional info on script: https://github.com/electron-userland/electron-builder/issues/4679

### Requirements

TBD

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

We build our Windows application by using the `Microsoft VS C++ Redist 2015-2022`. If one wants to run FF on a windows with an older version, it crashes instantly. 
With this PR, a script is added to the windows build, which checks if the current version is installed. 
- if it is -> continue
- if not -> download the official c++ redist installer for the newest version and execute it

Best how to test it:
Remove in your applications the Microsoft VS C++ Redistributable 2015-2022  , download windows build from https://github.com/iotaledger/firefly/actions/runs/2978247706 and install it. For a relative short moment you should see a second installation bar, which means it installs the missing dependency. After it's finished, check in your applications if you now have Microsoft VS C++ Redistributable 2015-2022  again installed. 
If not, then i messed up something,
If yes, then pls execute the installation script again, and this time the second installation bar should not appear.

## Changelog

```
- add nsis script on windows to install vs c++ dependency
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Closes #4250

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [x] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/webpack.config.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ const prod = mode === 'production'
 const hardcodeNodeEnv = typeof process.env.HARDCODE_NODE_ENV !== 'undefined'
 const SENTRY = process.env.SENTRY === 'true'
 const stage = process.env.STAGE || 'alpha'
+const os = process.env.OS || 'ubuntu-20.04'
```
Comment (maxwellmattryan) [inline packages/desktop/webpack.config.js:15]: Is this necessary? Would `process.platform` not work?

File: .github/workflows/build-desktop.yml [hunk 2]
```diff
@@ -120,6 +120,7 @@ jobs:
               working-directory: packages/desktop
               shell: bash
               env:
+                  OS: ${{ matrix.os }}
```
Comment (maxwellmattryan) [inline .github/workflows/build-desktop.yml:123]: May not be necessary if we can grab `process.platform` in the code.

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Have some comments / questions.

File: packages/desktop/webpack.config.js (hunk 1 above)
Comment (MarkNerdi) [reply packages/desktop/webpack.config.js:15]: nice catch

**Review by nicole-obrien — APPROVED** [review]

**Review by maxwellmattryan — APPROVED** [review]
LGTM 🔥

**Pull request merged**